### PR TITLE
Enabled cumsum and cumprod for bool tensors

### PR DESCRIPTION
**Pull request by izdeby**
```
a = torch.tensor([[True, False, True],
                  [False, False, False],
                  [True, True, True]])

>>> torch.cumsum(a, 0)
tensor([[1, 0, 1],
        [1, 0, 1],
        [2, 1, 2]])


>>> torch.cumsum(a, 1)
tensor([[1, 1, 2],
        [0, 0, 0],
        [1, 2, 3]])
```

Tested via unit tests.

### CODE REVIEW

File: torch/csrc/autograd/python_variable_indexing.cpp [hunk 1]
```diff
@@ -175,7 +175,7 @@ static Variable applySlicing(const Variable& self, PyObject* index, variable_lis
     } else if (THPVariable_Check(obj)) {
       auto& var = THPVariable_Unpack(obj);
       auto scalar_type = var.scalar_type();
-      if (var.dim() == 0 && (at::isIntegralType(scalar_type) || scalar_type == ScalarType::Bool)) {
+      if (var.dim() == 0 && at::isIntegralType(scalar_type)) {
```
Comment (gchanan) [inline torch/csrc/autograd/python_variable_indexing.cpp]: is this change related?  Where does this come up?  Also shouldn't there be a warning somewhere?

**Review by gchanan — APPROVED** [review]
looks good besides the indexing change.

File: torch/csrc/autograd/python_variable_indexing.cpp (hunk 1 above)
Comment (izdeby) [reply torch/csrc/autograd/python_variable_indexing.cpp]: i added bool to isIntegralType so this part became redundant now

**Comment by pietern** [comment]
@izdeby @gchanan Lint failure was introduced here. See failed Travis builds.
